### PR TITLE
https server for securecontext 

### PR DESCRIPTION
**Pull request by allopez7**
…nd logname into readdir

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -35,4 +38,15 @@ const listener = (): void => {
   app.listen(port, () => console.log(`app listening on port ${port}!`))
 }
 
+const server = https.createServer({
+  SNICallback: (domain, cb) => {
+    const adminUser = execSync('logname').toString()
+    fs.readdir(`/home/albertoe/\.acme\.sh`, (err, files)=>{
```
Comment (allopez7) [inline src/app.ts]: this is causing a warning currently, will update soon.

File: package.json [hunk 2]
```diff
@@ -36,6 +36,7 @@
     "express": "^4.17.1",
     "husky": "^3.0.3",
     "node-fetch": "^2.6.0",
+    "nodemon": "^1.19.2",
```
Comment (songz) [inline package.json]: remove.

File: src/app.ts [hunk 3]
```diff
@@ -35,4 +38,15 @@ const listener = (): void => {
   app.listen(port, () => console.log(`app listening on port ${port}!`))
 }
 
+const server = https.createServer({
+  SNICallback: (domain, cb) => {
+    const adminUser = execSync('logname').toString()
```
[the remark this answers is not in the document]
Comment (allopez7) [reply src/app.ts]: gives me 'root'

File: package.json (hunk 2 above)
Comment (allopez7) [reply package.json]: was throwing error unless I added.

Comment (songz) [reply package.json]: this should be a dev dependency. Could you remove, then run `yarn add nodemon -D`. In production, we will not be using nodemon

Comment (songz) [reply package.json]: since you fixed this in a different PR, could you remove this? https://github.com/garageScript/myProxy/pull/123

Comment (allopez7) [reply package.json]: ok

File: src/app.ts [hunk 4]
```diff
@@ -41,14 +45,38 @@ app.post('/login', (req, res) => {
 })
 
 const listener = (): void => {
-  const cyan = '\x1b[36m\u001b[1m%s\x1b[0m'
```
Comment (allopez7) [inline src/app.ts]: I did not delete these, pulled from main branch and this is the result.

File: src/app.ts [hunk 5]
```diff
@@ -41,14 +45,38 @@ app.post('/login', (req, res) => {
...
-  const red = '\x1b[31m\u001b[1m%s\x1b[0m'
```
Comment (songz) [inline src/app.ts]: why did you remove all these?

Comment (allopez7) [reply src/app.ts]: I did not remove them. Updated my branch and this happened.

File: src/app.ts [hunk 6]
```diff
@@ -51,4 +55,32 @@ const listener = (): void => {
...
+    SNICallback: (domain, cb) => {
+      // using whoami will return root because app is run with sudo
+      const adminUser = execSync('logname')
+        .toString()
+        .trim()
+      // escape characters required or readFileSync will not find file
+      const secureContext = tls.createSecureContext({
+        /* eslint-disable */
+        key: fs.readFileSync(
+          `/home/${adminUser}/\.acme\.sh/*\.${domain}/*\.${domain}\.key`
```
Comment (songz) [inline src/app.ts]: instead of `/home/${adminUser}`, could you do `~/` instead? Because for `root` user, the home path is `/root`

File: src/app.ts [hunk 7]
```diff
@@ -5,6 +5,10 @@ import crypto from 'crypto'
 import cookieParser from 'cookie-parser'
 import { adminRouter } from './admin/index'
 import { apiRouter } from './api/index'
+import https from 'https'
+import fs from 'fs'
+import { execSync } from 'child_process'
```
Comment (songz) [inline src/app.ts]: `execSync` is not used here.
